### PR TITLE
Enhance /exclude workflow and ProblemList insertions (#7340)

### LINKED ISSUE #7340: Improvements to /exclude workflow
Finding a few things that will improve the exclude workflow, https://github.com/adoptium/aqa-tests/blob/master/.github/workflows/autoTestPR.yml

- Consider improving the name of the workflow yml file so when trying to find it in .github/workflows, it is more obvious and related to what one might expect, autoTestPR.yml previously existed and also contains other logic, does it make sense to separate it
- Fix doc https://github.com/adoptium/aqa-tests/blob/master/docs/pages/userGuide.md#automatically-exclude-openjdk-test-cases-from-problemlist-files to point at correct reference doc (not openjdk_tests, but rather https://github.com/adoptium/aqa-tests/blob/master/openjdk/README.md#exclude-a-testcase).
- Attempt to insert the excluded testcases into the correct section of the ProblemList file, rather than appending to the end of the file (which can be the fallback behaviour if correct section can not be found).

### PR DESCRIPTION
**Pull request by Agaba-derrick**
- Extracted the exclude-openjdk-test job from autoTestPR.yml into a dedicated .github/workflows/exclude-openjdk-test.yml file to improve discoverability.

- Updated the documentation reference in docs/pages/userGuide.md to point to the correct openjdk/README.md#exclude-a-testcase anchor.

- Enhanced exclude_openjdk.py to insert test exclusions into the appropriate directory section within ProblemList files instead of blindly appending them to the bottom.

- Added unit tests in test_exclude_openjdk.py to validate the new section insertion logic across multiple edge cases.

Closes #7340

### CODE REVIEW

**Comment by Agaba-derrick** [comment]
CC: @smlambert  Kindly asking for a review

**Comment by karianna** [comment]
@Agaba-derrick Will need a rebase

**Review by karianna — DISMISSED** [review]

File: .github/workflows/exclude-openjdk-test.yml [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          COMMENTER: ${{ github.event.comment.user.login }}
+          ISSUE_AUTHOR: ${{ github.event.issue.user.login }}
+        run: |
+          # Allow if commenter is the issue author
+          if [ "$COMMENTER" = "$ISSUE_AUTHOR" ]; then
+            echo "✅ Commenter is the issue author"
+            exit 0
+          fi
```
[the remark this answers is not in the document]
Comment (sophia-guo) [reply .github/workflows/exclude-openjdk-test.yml]: Any external user can open an issue. This grants any external user can auto create PRs, but PRs won't be merged unless it's reviewed by comitters.

We do have a list of user signed ECA, if we can use that list as permission check the risk will be decreased. @smlambert is it possible to get that list?

[the remark this answers is not in the document]
Comment (smlambert) [reply .github/workflows/exclude-openjdk-test.yml]: I do not think it is a requirement for external users to call this workflow, it can be reserved for use by AQAvit committers.

[the remark this answers is not in the document]
Comment (sophia-guo) [reply .github/workflows/exclude-openjdk-test.yml]: > I do not think it is a requirement for external users to call this workflow, it can be reserved for use by AQAvit committers.

@Agaba-derrick could you update to limit the permission to AQAvit committers? Current rule is both issue author and AQAvit committers have the permission,
